common/types.go

```diff
@@ -78,6 +78,9 @@ var (
 
 	// ErrBadRequest is returned when we get a 400 response from the provider.
 	ErrBadRequest = errors.New("bad request")
+
+	// ErrEmptyResponse is returned when the jsonResponse is nil
+	ErrEmptyResponse = errors.New("empty http response error")
```

Comment:
Can you name this as `ErrEmptyJSONHTTPResponse` and change the error message to `empty json http response`? You will also need to fix linting errors.